obspy/clients/fdsn/client.py

```diff
@@ -555,6 +555,26 @@ def get_events(self, starttime=None, endtime=None, minlatitude=None,
             data_stream.close()
             return cat
 
+    def get_availability(self, network=None, station=None, location=None, 
+                         channel=None, starttime=None, endtime=None, 
+                         quality=None, minimumlength=None, longestonly=None, 
+                         format=None, filename=None, **kwargs):
+        if "availability" not in self.services:
+            msg = "The current client does not have an availability service."
+            raise ValueError(msg)
+
+        locs = locals()
+        setup_query_dict('availability', locs, kwargs)
+
+        url = self._create_url_from_parameters(
+            "availability", DEFAULT_PARAMETERS['availability'], kwargs)
+
+        availability = self._download(url, return_string=True).decode()
+        lines = [line.split() for line in availability.strip().split('\n')[1:]] # skip header 
+        extents = [(line[0], line[1], line[2], line[3], UTCDateTime(line[6]), UTCDateTime(line[7])) for line in lines]
```

Comment:
I think how the return type looks like is one of the most important things to decide, but there might not be more sophisticated ways to return this I guess, so this simple structure might be fine

Reply:
yes. I am looking at [this](https://docs.obspy.org/packages/autogen/obspy.clients.earthworm.client.Client.get_availability.html) and also at the API specification [here](https://docs.obspy.org/packages/autogen/obspy.clients.earthworm.client.Client.get_availability.html) as examples.

Relatedly, I notice that there are optional parameters in the [specification](https://www.fdsn.org/webservices/fdsnws-availability-1.0.pdf) ('merge', 'show') that change the number of fields included in the results from FDSN.